### PR TITLE
Fix pdf upload and pdf download from search results

### LINKED ISSUE #6982: App crashes on pdf upload.
Per Mattermost guidelines, GitHub issues are for bug reports: <http://www.mattermost.org/filing-issues/>.

For troubleshooting see: http://forum.mattermost.org/.
For feature proposals see: http://www.mattermost.org/feature-requests/

If you've found a bug--something appears unintentional--please follow these steps:

1. Confirm you’re filing a new issue. [Search existing tickets in Jira](https://mattermost.atlassian.net/jira/software/c/projects/MM/issues/) to ensure that the ticket does not already exist.
2. Confirm your issue does not involve security. Otherwise, please see our [Responsible Disclosure Policy](https://about.mattermost.com/report-security-issue/).
3. [File a new issue](https://github.com/mattermost/mattermost-mobile/issues/new) using the format below. Mattermost will confirm steps to reproduce and file in Jira, or ask for more details if there is trouble reproducing it. If there's already an existing bug in Jira, it will be linked back to the GitHub issue so you can track when it gets fixed.

#### Summary
Bug report in one concise sentence

### Environment Information
  - Device Name: Pixel 3
  - OS Version: 
  - Mattermost App Version:2.0.0
  - Mattermost Server Version:

#### Steps to reproduce
How can we reproduce the issue (what version are you using?)
-Open any dm or channel
-Click on attachment icon (Works fine on Pictures but crashes on pdf)
-Select any PDf from mobile.
-App will crash

#### Expected behavior
How can we reproduce the issue (what version are you using?)
-Open any dm or channel
-Click on attachment icon (Works fine on Pictures but crashes on pdf)
-Select any PDf from mobile.
-It should attach the selected file

#### Observed behavior (that appears unintentional)
What did you see happen? Please include relevant error messages, screenshots and/or video recordings.

#### Possible fixes
If you can, link to the line of code that might be responsible for the problem

https://user-images.githubusercontent.com/111849328/212833298-3608ef6b-1d0d-4881-bdcd-88c74f3f0917.mp4

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
This PR fixes two problems:
- When trying to upload a pdf file, the app crashes.
- When trying to download a pdf file directly from the search results, an error appears

DISCLAIMER: for the uploading pdf file, I just piggybacked a fix from @enahum , so all credit for him.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49707
Fix #6982 

#### Release Note
```release-note
Fix issue with uploading certain files.
Fix issue when downloading certain files from the search results.
```

### CODE REVIEW

File: app/utils/file/file_picker/index.ts [hunk 1]
```diff
@@ -272,7 +276,7 @@ export default class FilePickerUtil {
 
         if (hasPermission) {
             try {
-                const docResponse = (await DocumentPicker.pick({allowMultiSelection, type: [fileType]}));
+                const docResponse = (await DocumentPicker.pick({allowMultiSelection, type: [fileType], copyTo: 'cachesDirectory'}));
```
Comment (avinashlng1080) [inline app/utils/file/file_picker/index.ts:279]: What is the below a string and not a constant like `FileSystem.CachesDirectoryPath` ?

```
 copyTo: 'cachesDirectory'}));
```

**Review by avinashlng1080 — COMMENTED** [review]
Ready to approve.

Which part is Elias fix and there's a question regarding 'cachesDirectory' below?

File: app/utils/file/file_picker/index.ts (hunk 1 above)
Comment (larkox) [reply app/utils/file/file_picker/index.ts:279]: @enahum :point_up:

Comment (enahum) [reply app/utils/file/file_picker/index.ts:279]: The value is a string as is required by the type definition, being a type definition you only have two valid options

**Comment by jgilliam17** [comment]
Thanks @larkox 
Uploading PDF is working as expected
I am seeing an issue when downloading PDF from search. 
- Steps:
Click on file item 3dot menu in search results, select download, select file location and save
Observed: App locks up, unresponsive to tapping, I had to force quit and reopen. I've sent you an email via "report a problem " link in Settings, in case there could be some useful info shared there.

**Review by avinashlng1080 — APPROVED** [review]

**Review by jfrerich — APPROVED** [review]

**Comment by enahum** [comment]
@amyblais @esethna  this PR fixes a crash when trying to upload a pdf file, this should be cherry-pick for 2.0.1 imo

**Comment by enahum** [comment]
@larkox conflicts

**Comment by enahum** [comment]
@jgilliam17 can you please test this PR? Thank you

**Comment by jgilliam17** [comment]
@enahum yes, I re-test this again. I had an issue, Daniel was looking into, where app became unresponsive after pdf download. Will let you know if that is still happening on iOS.

**Comment by jgilliam17** [comment]
I am still seeing app locking up after pdf download on iPhone 13, v16.2
Steps:
1. search for pdf file
2. tap 3 dot menu and select Download
3. Select Save without changing folder/location
Observed: app is unresponsive, has to be closed and reopened

**Comment by enahum** [comment]
@jgilliam17 it should be fixed now

**Review by jgilliam17 — APPROVED** [review]
Thanks @enahum - that did it!
I can upload and download PDFs from search results - working as expected

**Comment by enahum** [comment]
I'm on it

**Comment by enahum** [comment]
done

**Pull request merged**